src/main/java/com/microsoft/sqlserver/jdbc/SQLServerColumnEncryptionAzureKeyVaultProvider.java

```diff
@@ -91,69 +93,37 @@ public String getName() {
      *        Identifier of the client requesting the token.
      * @param clientKey
      *        Key of the client requesting the token.
+     * @param tenantId
+     *        The tenant ID of the application.
      * @throws SQLServerException
      *         when an error occurs
      */
-    public SQLServerColumnEncryptionAzureKeyVaultProvider(String clientId, String clientKey) throws SQLServerException {
-        credentials = new KeyVaultCredential(clientId, clientKey);
-        keyVaultClient = new KeyVaultClient(credentials);
-    }
-
-    /**
-     * Constructs a SQLServerColumnEncryptionAzureKeyVaultProvider with a callback function to authenticate to AAD and
-     * an executor service.. This is used by KeyVaultClient at runtime to authenticate to Azure Key Vault.
-     * 
-     * This constructor is present to maintain backwards compatibility with 6.0 version of the driver. Deprecated for
-     * removal in next stable release.
-     * 
-     * @param authenticationCallback
-     *        - Callback function used for authenticating to AAD.
-     * @param executorService
-     *        - The ExecutorService, previously used to create the keyVaultClient, but not in use anymore. - This
-     *        parameter can be passed as 'null'
-     * @throws SQLServerException
-     *         when an error occurs
-     */
-    @Deprecated
-    public SQLServerColumnEncryptionAzureKeyVaultProvider(
-            SQLServerKeyVaultAuthenticationCallback authenticationCallback,
-            ExecutorService executorService) throws SQLServerException {
-        this(authenticationCallback);
-    }
-
-    /**
-     * Constructs a SQLServerColumnEncryptionAzureKeyVaultProvider with a callback function to authenticate to AAD. This
-     * is used by KeyVaultClient at runtime to authenticate to Azure Key Vault.
-     * 
-     * @param authenticationCallback
-     *        - Callback function used for authenticating to AAD.
-     * @throws SQLServerException
-     *         when an error occurs
-     */
-    public SQLServerColumnEncryptionAzureKeyVaultProvider(
-            SQLServerKeyVaultAuthenticationCallback authenticationCallback) throws SQLServerException {
-        if (null == authenticationCallback) {
+    public SQLServerColumnEncryptionAzureKeyVaultProvider(String clientId,
+                                                          String clientKey,
+                                                          String tenantId) throws SQLServerException {
```

Comment:
removing `SQLServerColumnEncryptionAzureKeyVaultProvider(String clientId, String clientKey)` is a breaking change. Can we keep it and get tenantId elsewhere?